frontend/src/components/PrivateRoute.js

```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { Navigate, Outlet } from 'react-router-dom';
```

Comment:
you need to create /routes and inside /AuthRoutes /PublicRoutes and PrivateRoutes
also for your api call create a folder /api inside src like UserApi.js